src/snowflake/connector/result_set.py

```diff
@@ -61,12 +64,40 @@ def result_set_iterator(
     Just like ``ResultBatch`` iterator, this might yield an ``Exception`` to allow users
     to continue iterating through the rest of the ``ResultBatch``.
     """
-    for it in pre_iter:
-        for element in it:
-            yield element
-    for it in post_iter:
-        for element in it.create_iter(**kw):
-            yield element
+
+    yield from first_batch_iter
```

Comment:
I'm sorry I don't remember anymore if this is in line with how pre-fetching worked in the old implementation. Should this not be after we submit the next 4 batches to `pool`?

Reply:
You are right, the `pre_init` in the chunk downloader pre-fetches the first 4 batches before any yielding. I'll update this.